### ISSUE TITLE
iOS 14 “Not responding”

### ISSUE THREAD

**Issue by durosity** [Colorado4Wheeler/HomeKit-Bridge#138]
I'm perpetually getting "updating.." then "not responding" messages with all HKB devices running iOS 14.  I was getting the same thing with my stand alone homebridge server (used for harmony hubs, etc) until I updated it to the latest release of homebridge, so I'm presuming something that's changed between the older releases bundled in with HKB and the current one that helps with iOS 14 compatibility.  I know you've said you're really busy at the moment but is there any plans to bring it up to date?  

Just for reference I've also tried the following troubleshooting steps:

Restored iPhone
Rebuilt the HomeKit folder using advanced plugin tools
Deleted the home and setup from scratch
Deleted all references to HKB in indigo and rebuilt from scratch
Started completely from scratch on a new Apple ID on a separate iPhone and setup a new DB on a brand new indigo install on a completely different Mac, just to take every variable I could think of out of the equation!  Alas.. no joy!

**Comment by Colorado4Wheeler** [comment]
My iOS 14 is working great, but I can work on building out a new version that has the latest HomeKit support in it.

**Comment by durosity** [comment]
Yeah, I've not seen anyone else on the indigo forums specify this exact issue, although there has been mumbling over time of other things.. the only reason I was thinking a newer release might help is my own standalone homebridge update fixing the same issue for that, but that's not to say it fixed something else coincidentally!  However if you can update it with relative ease it would be greatly appreciate!  Don't stress it though if you're busy.. you owe me nothing!

**Comment by Colorado4Wheeler** [comment]
I’m out of town this week so I can’t do anything but we can chat next week and see what might be possible.

**Comment by durosity** [comment]
Cool!  It was nice to hear from you.. until you posted a little while ago I thought you were gone forever!

**Comment by Colorado4Wheeler** [comment]
Haha, no.  There are a few reasons why I haven’t been active on Indigo for a while.  The first and foremost is that my HomeKit is 99% of how I use my home automation and it’s been working perfectly for me, there was never a reason for me to do more.  Second is not only has COVID kept me busier than I’ve been in years but I bought a couple more businesses and THOSE have kept me hopping. 

I’m still around, and now I have a reason to mess with HomeKit again :P.  Of course I know it is a rabbit hole, once I start I’m going to want to do more cool stuff with it, so I may be around much more now!

**Comment by durosity** [comment]
Well it's good to have things to keep you busy.  I don't know how you manage to have time to juggle all these businesses.. I feel like I don't even have time to do my one single job these days!  

And.. that makes me excited.. when you start getting ideas plebs like me start getting features we'd never dreamed of 😂

**Comment by Colorado4Wheeler** [comment]
LOL!  Yea, that is why I haven’t headed down that path yet, I also get super excited when I start adding features, which leads to more features, more excitement and before I know it then I’m spending two solid months making the plugin better.  It is a slippery slope :).

Do you have a new updated compilation of the HomeKit library ready to go?

**Comment by durosity** [comment]
I'm afraid not, I wouldn't even know where to start with that!  Unfortunately that's the bit where my knowledge just stops dead... as much as I really should know it. 😕

**Comment by Colorado4Wheeler** [comment]
I thought you were the one that compiled the original library, I’ll have to see if I can touch base with the OTHER Mike to get that unix compilation.

**Comment by durosity** [comment]
Ahh no no, I'm the stupid Mike.  😂

**Comment by Colorado4Wheeler** [comment]
I need ‘smart Mike’ to assist with that part, I’ll reach out.  😜

**Comment by gt3mike** [comment]
I'm also Mike, but I'm definitely not that smart. 😄  FWIW, so far this seems to be working with iOS 14 for me (though my usage is limited). 

I'm just starting to play with this plugin, having just removed Alexa from my home. A bit bumpy, but I'm making progress. I'm reading through the issues and will create one for something I just experienced if I don't see it listed already.

**Comment by durosity** [comment]
I'm Mike and so's my wife!

Yeah I'm convinced it's from the quantity of devices I punish (about 150 inc some that relate to activity groups) but it was and continues to work with no problems with iOS 13.  I've actually got a stable work around which is an old iPhone SE locked in kiosk mode on the home app.. for some reason that seems to keep the states fresh so it doesn't have to poll too many at once when it's opened on my normal iPhone.  However it's not exactly a long term fix, so I'm hopeful that an update to the underlying homebridge version will help as I've added 99 dummy devices to my stand alone version of HB and it still works fine.. ¯\_(ツ)_/¯

**Comment by webdeck** [comment]
"smart Mike" here.  I have a rebuild of the latest version of homebridge, but much has changed.  Some caveats of the new build:

- Minimum macOS version is now 10.14 (up from 10.7) due to new minimum requirements for ffmpeg
- New config.json requirement: must include mdns section to specify which IP address to listen on
- Added getips script to return the available IP addresses so that you can add that to your configuration

I have done some testing in my home and it seems to be working fine, but since your plugin overwrites config.json, I lose the mdns entry every restart, which breaks my config, since my server has multiple IP addresses and it chooses the wrong one by default.

**Comment by Colorado4Wheeler** [comment]
Can you send me a sample config.json to see how that section needs to be updated?

**Comment by webdeck** [comment]
Yes, I just sent you an email with more details and the build.
